src/bcf/mod.rs

```diff
@@ -22,10 +22,41 @@ pub use bcf::buffer::RecordBuffer;
 
 
 /// Redefinition of corresponding `#define` in `vcf.h.`.
-#[allow(non_upper_case_globals)]
 pub const GT_MISSING: i32 = 0;
 
 
+/// A trait for a BCF reader with a read method.
+pub trait Read: Sized {
+    /// Read the next record.
+    ///
+    /// # Arguments
+    /// * record - an empty record, that can be created with `bcf::Reader::empty_record`.
+    fn read(&mut self, record: &mut record::Record) -> Result<(), ReadError>;
+
+    /// Return an iterator over all records of the VCF/BCF file.
+    fn records(&mut self) -> Records<Self>;
+
+    /// Return the header.
+    fn header(&self) -> Rc<HeaderView>;
```

Comment:
Why not returning `&HeaderView`?

Reply:
Ah, I think now I understand the problem with empty_record. In that case, I think it is better to keep the API clean, return &HeaderView, and leave the implementation of empty_record to the implementing structs.

Reply:
Because `bcf::record::Record::new` wants a `Rc<HeaderView>`...